CONTRIBUTING.md

```diff
@@ -357,19 +477,71 @@ Beside general good practices for regular expresions, keep in mind:
-| WKN           | A111X9       | `\\w+`              | `[A-Z0-9]{6}` |
-| Amount        | 751,68       | `[\\d,.]+`          | `[\\.,\\d]+` |
-|               |              |                     | `[\\.\\d]+,[\\d]{2}` |
-|               | 74'120.00    | `[\\d.']+`          | `[\\.'\\d]+` |
-|               | 20 120.00    | `[\\d.\\s]+`        | `[\\.\\d\\s]+` |
-| Currency      | EUR          | `\\w+`              | `[A-Z]{3}` |
-|               |              |                     | `[\\w]{3}` |
-| Currency      | € or $       | `\\D`               | `\\p{Sc}` |
+Keep in mind that the regular expressions work against text that is automatically created from PDF files. Due to the nature of the process, there can always be slight differences in the text files. The following table collects the regular expressions that **worked well** to match typical values.
+
+| Value	| Example	| Not Helpful	| Works Well	|
+| :-----------	| :-----------	| :------------	| :-----------	|
+| Date			| 01.01.1970	| `\\d+.\\d+.\\d{4}`	| `[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}`			|
+|				| 1.1.1970		| `\\d+.\\d+.\\d{4}`	| `[\\d]{1,2}\\.[\\d]{1,2}\\.[\\d]{4}`	|
+| Time			| 12:01			| `\\d+:\\d+`			| `[\\d]{2}\\:[\\d]{2}}`						|
+| ISIN			| IE00BKM4GZ66	| `\\w+`					| `[A-Z]{2}[A-Z0-9]{9}[0-9]`					|
+| WKN			| A111X9		| `\\w+`					| `[A-Z0-9]{6}`									|
+| Amount		| 751,68		| `[\\d,.]+`				| `[\\.,\\d]+`									|
+|				|				|							| `[\\.\\d]+,[\\d]{2}`							|
+|				| 74'120.00		| `[\\d.']+`				| `[\\.'\\d]+`									|
+|				| 20 120.00		| `[\\d.\\s]+`			| `[\\.\\d\\s]+`								|
+| Currency      | EUR			| `\\w+`					| `[A-Z]{3}`										|
+|				|				|							| `[\\w]{3}`										|
+| Currency      | € or $		| `\\D`					| `\\p{Sc}`										|
+
+
+## Trade calendar
+
+Using the application menu, the user can select a trading calendar accordingly globally or for each individual security.
+The calendar takes into account the weekends and regional holidays when there is no stock exchange trading.
+The trading-free days of the stock exchange itself, if it is a stock exchange calendar, are also taken into account.
+The individual trading-free days are stored in a HashMap and made available for further processing, e.g. reporting period, performance index and so on.
+
+
+### Source location
+
+Trade calendar Manager: `name.abuchen.portfolio/src/name/abuchen/portfolio/util/TradeCalendarManager.java`
+
+Trade calendar Class: `name.abuchen.portfolio/src/name/abuchen/portfolio/util/TradeCalendar.java`
+
+Holiday types: `name.abuchen.portfolio/src/name/abuchen/portfolio/util/HolidayType.java`
+
+Holiday class: `name.abuchen.portfolio/src/name/abuchen/portfolio/util/Holiday.java`
+
+Holiday name: `name.abuchen.portfolio/src/name/abuchen/portfolio/util/HolidayName.java`
+
+Tests: `name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java`
+
+
+### Anatomy of a trade calendar
+
+The structure of the trade calendars is as follows:
+
```

Comment:
Again, I think at most we can point out to common things to contribute (example: trade calendar). This just blows up the contributing guide. And if somebody does not understand the code, they'll also not be able to contribute meaningful code anyways